### PR TITLE
Support Post Date in Export of integration xero api change

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: apps/xero/models.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def get_transaction_date(expense_group: ExpenseGroup) -> str:
         return expense_group.description['verified_at']
     elif 'last_spent_at' in expense_group.description and expense_group.description['last_spent_at']:
         return expense_group.description['last_spent_at']
+    elif 'posted_at' in expense_group.description and expense_group.description['posted_at']:
+        return expense_group.description['posted_at']
```
Comment (ashwin1111) [inline apps/xero/models.py:60]: test coverage is not reaching this line it seems, can you add test to cover this?

File: apps/workspaces/apis/export_settings/serializers.py [hunk 2]
```diff
@@ -133,7 +133,6 @@ def update(self, instance, validated):
         expense_group_settings['expense_state'] = expense_group_settings_instance.expense_state
         expense_group_settings['reimbursable_expense_group_fields'] = expense_group_settings_instance.reimbursable_expense_group_fields
         expense_group_settings['corporate_credit_card_expense_group_fields'] = expense_group_settings_instance.corporate_credit_card_expense_group_fields
-        expense_group_settings['ccc_export_date_type'] = expense_group_settings_instance.ccc_export_date_type
```
Comment (ashwin1111) [inline apps/workspaces/apis/export_settings/serializers.py:136]: why is this removed?

File: apps/xero/models.py [hunk 3]
```diff
@@ -282,8 +284,6 @@ def create_bank_transaction(expense_group: ExpenseGroup, map_merchant_to_contact
                 value__iexact=merchant, attribute_type='CONTACT', workspace_id=expense_group.workspace_id
             ).first()
 
-            expense_group.description['spent_at'] = expense.spent_at.strftime('%Y-%m-%d')
-            expense_group.save()
```
Comment (ashwin1111) [inline apps/xero/models.py:286]: why is this removed?

File: apps/xero/models.py [hunk 4]
```diff
@@ -323,6 +323,7 @@ def create_bank_transaction(expense_group: ExpenseGroup, map_merchant_to_contact
                 'transaction_date': get_transaction_date(expense_group),
             }
         )
+        print("bank_transaction_object", bank_transaction_object.__dict__)
```
Comment (ashwin1111) [inline apps/xero/models.py:326]: remove loggers pls

File: apps/workspaces/apis/export_settings/serializers.py (hunk 2 above)
Comment (DhaaraniCIT) [reply apps/workspaces/apis/export_settings/serializers.py:136]: before there was only spent_at, but now we when we have this, the change in the ccc_export_date_type is not updated, the export Api is returning only spend-at as ccc export date type

File: apps/xero/models.py (hunk 3 above)
Comment (DhaaraniCIT) [reply apps/xero/models.py:286]: before there was only spent_at, but now when we have this, the change in the ccc_export_date_type, because this code even though we have other types it will add spend_at in the description, which will not return the expected result

File: apps/workspaces/apis/export_settings/serializers.py (hunk 2 above)
Comment (ashwin1111) [reply apps/workspaces/apis/export_settings/serializers.py:136]: was this done in QBO too?

Comment (DhaaraniCIT) [reply apps/workspaces/apis/export_settings/serializers.py:136]: no

File: apps/xero/models.py (hunk 3 above)
Comment (DhaaraniCIT) [reply apps/xero/models.py:286]: This was done in QBO too

Comment (DhaaraniCIT) [reply apps/xero/models.py:286]: this was done because the spend_at is no longer the default setting, for the transaction date

File: apps/workspaces/apis/export_settings/serializers.py (hunk 2 above)
Comment (ashwin1111) [reply apps/workspaces/apis/export_settings/serializers.py:136]: got it

**Review by ashwin1111 — APPROVED** [review]
